src/hotspot/os/linux/cgroupUtil_linux.hpp

```diff
@@ -32,6 +32,12 @@ class CgroupUtil: AllStatic {
 
   public:
     static int processor_count(CgroupCpuController* cpu, int host_cpus);
+    // Iterate over the memory controller hierarchy adjusting the path to the
+    // smallest observable limit (if any)
+    static CgroupMemoryController* adjust_controller(CgroupMemoryController* m);
+    // Iterate over the cpu controller hierarchy adjusting the path to the
+    // smallest observable limit (if any)
+    static CgroupCpuController* adjust_controller(CgroupCpuController* c);
```

Comment:
We never return a new controller, but always the old one. I think I would simplify and clarify the API and return void. I would not allow for API chaining here, the memory ownerships are difficult enough to follow as it is. Could also return boolean for success=limit found. And comment should make it clear we operate on the given controller.

Proposal:

```
Given a memory controller, adjust its path to a point in the hierarchy 
that represents the closest memory limit. Returns true if such limit was
 found, false if not.
```

Reply:
OK. I've changed the API to return `void` and fixed the comment.